BepInEx.Preloader.Core/Patching/AssemblyPatcher.cs

```diff
@@ -150,69 +157,155 @@ public void AddPatchersFromDirectory(string directory)
 
-                    foreach (var method in methods)
-                    {
-                        var targetAssemblies = MetadataHelper.GetAttributes<TargetAssemblyAttribute>(method);
-                        var targetTypes = MetadataHelper.GetAttributes<TargetTypeAttribute>(method);
-
-                        if (targetAssemblies.Length == 0 && targetTypes.Length == 0)
-                            continue;
-
-                        var parameters = method.GetParameters();
-
-                        if (parameters.Length < 1 || parameters.Length > 2
-                                                     // Next few lines ensure that the first parameter is AssemblyDefinition and does not have any
-                                                     // target type attributes, and vice versa
-                                                  || !(
-                                                          parameters[0].ParameterType == typeof(AssemblyDefinition)
-                                                       || parameters[0].ParameterType ==
-                                                          typeof(AssemblyDefinition).MakeByRefType()
-                                                       && targetTypes.Length == 0
-                                                       || parameters[0].ParameterType == typeof(TypeDefinition)
-                                                       && targetAssemblies.Length == 0
-                                                      )
-                                                  || parameters.Length == 2 &&
-                                                     parameters[1].ParameterType != typeof(string)
-                                                  || method.ReturnType != typeof(void) &&
-                                                     method.ReturnType != typeof(bool)
-                           )
-                        {
-                            Logger
-                                .Log(LogLevel.Warning,
-                                     $"Skipping method [{method.FullDescription()}] as it is not a valid patcher method");
-                            continue;
-                        }
+        PatcherContext.PatchDefinitions.AddRange(sortedPatchers);
+        Logger.Log(LogLevel.Info, $"Loaded {sortedPatchers.Count} patcher{(sortedPatchers.Count > 0 ? "s" : "")}");
+    }
 
-                        void AddDefinition(PatchDefinition definition)
-                        {
-                            Logger.Log(LogLevel.Debug, $"Discovered patch [{definition.FullName}]");
-                            sortedPatchers.Add(definition);
-                        }
+    private Assembly PatcherProvidersAssemblyResolver(object sender, ResolveEventArgs args)
+    {
+        foreach (BasePatcherProvider provider in providers)
+        {
+            var ass = provider.ResolveAssembly(args.Name);
+            if (ass != null)
+                return ass;
+        }
```

Comment:
It might make sense to wrap the ResolveAssembly in a trycatch in case there's a single faulty provider so it doesn't bring down everything else with it,.

Reply:
Done